{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import idna\n",
    "# import dnspython\n",
    "import urllib.parse\n",
    "import pymongo\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "ConfigurationError",
     "evalue": "A DNS label is > 63 octets long.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mUnicodeError\u001b[0m                              Traceback (most recent call last)",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\name.py:211\u001b[0m, in \u001b[0;36mIDNA2003Codec.encode\u001b[1;34m(self, label)\u001b[0m\n\u001b[0;32m    210\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m--> 211\u001b[0m     \u001b[39mreturn\u001b[39;00m encodings\u001b[39m.\u001b[39;49midna\u001b[39m.\u001b[39;49mToASCII(label)\n\u001b[0;32m    212\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mUnicodeError\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\encodings\\idna.py:74\u001b[0m, in \u001b[0;36mToASCII\u001b[1;34m(label)\u001b[0m\n\u001b[0;32m     73\u001b[0m \u001b[39m# Step 2: nameprep\u001b[39;00m\n\u001b[1;32m---> 74\u001b[0m label \u001b[39m=\u001b[39m nameprep(label)\n\u001b[0;32m     76\u001b[0m \u001b[39m# Step 3: UseSTD3ASCIIRules is false\u001b[39;00m\n\u001b[0;32m     77\u001b[0m \u001b[39m# Step 4: try ASCII\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\encodings\\idna.py:50\u001b[0m, in \u001b[0;36mnameprep\u001b[1;34m(label)\u001b[0m\n\u001b[0;32m     49\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39many\u001b[39m(stringprep\u001b[39m.\u001b[39min_table_d2(x) \u001b[39mfor\u001b[39;00m x \u001b[39min\u001b[39;00m label):\n\u001b[1;32m---> 50\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mUnicodeError\u001b[39;00m(\u001b[39m\"\u001b[39m\u001b[39mViolation of BIDI requirement 2\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m     51\u001b[0m \u001b[39m# 3) If a string contains any RandALCat character, a\u001b[39;00m\n\u001b[0;32m     52\u001b[0m \u001b[39m# RandALCat character MUST be the first character of the\u001b[39;00m\n\u001b[0;32m     53\u001b[0m \u001b[39m# string, and a RandALCat character MUST be the last\u001b[39;00m\n\u001b[0;32m     54\u001b[0m \u001b[39m# character of the string.\u001b[39;00m\n",
      "\u001b[1;31mUnicodeError\u001b[0m: Violation of BIDI requirement 2",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mLabelTooLong\u001b[0m                              Traceback (most recent call last)",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:89\u001b[0m, in \u001b[0;36m_SrvResolver._resolve_uri\u001b[1;34m(self, encapsulate_errors)\u001b[0m\n\u001b[0;32m     88\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m---> 89\u001b[0m     results \u001b[39m=\u001b[39m _resolve(\n\u001b[0;32m     90\u001b[0m         \u001b[39m\"\u001b[39;49m\u001b[39m_\u001b[39;49m\u001b[39m\"\u001b[39;49m \u001b[39m+\u001b[39;49m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__srv \u001b[39m+\u001b[39;49m \u001b[39m\"\u001b[39;49m\u001b[39m._tcp.\u001b[39;49m\u001b[39m\"\u001b[39;49m \u001b[39m+\u001b[39;49m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__fqdn, \u001b[39m\"\u001b[39;49m\u001b[39mSRV\u001b[39;49m\u001b[39m\"\u001b[39;49m, lifetime\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__connect_timeout\n\u001b[0;32m     91\u001b[0m     )\n\u001b[0;32m     92\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mException\u001b[39;00m \u001b[39mas\u001b[39;00m exc:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:43\u001b[0m, in \u001b[0;36m_resolve\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m     41\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mhasattr\u001b[39m(resolver, \u001b[39m\"\u001b[39m\u001b[39mresolve\u001b[39m\u001b[39m\"\u001b[39m):\n\u001b[0;32m     42\u001b[0m     \u001b[39m# dnspython >= 2\u001b[39;00m\n\u001b[1;32m---> 43\u001b[0m     \u001b[39mreturn\u001b[39;00m resolver\u001b[39m.\u001b[39;49mresolve(\u001b[39m*\u001b[39;49margs, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n\u001b[0;32m     44\u001b[0m \u001b[39m# dnspython 1.X\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:1368\u001b[0m, in \u001b[0;36mresolve\u001b[1;34m(qname, rdtype, rdclass, tcp, source, raise_on_no_answer, source_port, lifetime, search)\u001b[0m\n\u001b[0;32m   1359\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"Query nameservers to find the answer to the question.\u001b[39;00m\n\u001b[0;32m   1360\u001b[0m \n\u001b[0;32m   1361\u001b[0m \u001b[39mThis is a convenience function that uses the default resolver\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1365\u001b[0m \u001b[39mparameters.\u001b[39;00m\n\u001b[0;32m   1366\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m-> 1368\u001b[0m \u001b[39mreturn\u001b[39;00m get_default_resolver()\u001b[39m.\u001b[39mresolve(\n\u001b[0;32m   1369\u001b[0m     qname,\n\u001b[0;32m   1370\u001b[0m     rdtype,\n\u001b[0;32m   1371\u001b[0m     rdclass,\n\u001b[0;32m   1372\u001b[0m     tcp,\n\u001b[0;32m   1373\u001b[0m     source,\n\u001b[0;32m   1374\u001b[0m     raise_on_no_answer,\n\u001b[0;32m   1375\u001b[0m     source_port,\n\u001b[0;32m   1376\u001b[0m     lifetime,\n\u001b[0;32m   1377\u001b[0m     search,\n\u001b[0;32m   1378\u001b[0m )\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:1331\u001b[0m, in \u001b[0;36mget_default_resolver\u001b[1;34m()\u001b[0m\n\u001b[0;32m   1330\u001b[0m \u001b[39mif\u001b[39;00m default_resolver \u001b[39mis\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m-> 1331\u001b[0m     reset_default_resolver()\n\u001b[0;32m   1332\u001b[0m \u001b[39massert\u001b[39;00m default_resolver \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:1344\u001b[0m, in \u001b[0;36mreset_default_resolver\u001b[1;34m()\u001b[0m\n\u001b[0;32m   1343\u001b[0m \u001b[39mglobal\u001b[39;00m default_resolver\n\u001b[1;32m-> 1344\u001b[0m default_resolver \u001b[39m=\u001b[39m Resolver()\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:856\u001b[0m, in \u001b[0;36mBaseResolver.__init__\u001b[1;34m(self, filename, configure)\u001b[0m\n\u001b[0;32m    845\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"*filename*, a ``str`` or file object, specifying a file\u001b[39;00m\n\u001b[0;32m    846\u001b[0m \u001b[39min standard /etc/resolv.conf format.  This parameter is meaningful\u001b[39;00m\n\u001b[0;32m    847\u001b[0m \u001b[39monly when *configure* is true and the platform is POSIX.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    853\u001b[0m \u001b[39mon Windows systems.)\u001b[39;00m\n\u001b[0;32m    854\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m--> 856\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mreset()\n\u001b[0;32m    857\u001b[0m \u001b[39mif\u001b[39;00m configure:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:866\u001b[0m, in \u001b[0;36mBaseResolver.reset\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    864\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"Reset all resolver configuration to the defaults.\"\"\"\u001b[39;00m\n\u001b[1;32m--> 866\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mdomain \u001b[39m=\u001b[39m dns\u001b[39m.\u001b[39mname\u001b[39m.\u001b[39mName(dns\u001b[39m.\u001b[39;49mname\u001b[39m.\u001b[39;49mfrom_text(socket\u001b[39m.\u001b[39;49mgethostname())[\u001b[39m1\u001b[39m:])\n\u001b[0;32m    867\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mdomain) \u001b[39m==\u001b[39m \u001b[39m0\u001b[39m:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\name.py:963\u001b[0m, in \u001b[0;36mfrom_text\u001b[1;34m(text, origin, idna_codec)\u001b[0m\n\u001b[0;32m    961\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m is_all_ascii(text):\n\u001b[0;32m    962\u001b[0m     \u001b[39m# Some codepoint in the input text is > 127, so IDNA applies.\u001b[39;00m\n\u001b[1;32m--> 963\u001b[0m     \u001b[39mreturn\u001b[39;00m from_unicode(text, origin, idna_codec)\n\u001b[0;32m    964\u001b[0m \u001b[39m# The input is all ASCII, so treat this like an ordinary non-IDNA\u001b[39;00m\n\u001b[0;32m    965\u001b[0m \u001b[39m# domain name.  Note that \"all ASCII\" is about the input text,\u001b[39;00m\n\u001b[0;32m    966\u001b[0m \u001b[39m# not the codepoints in the domain name.  E.g. if text has value\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    970\u001b[0m \u001b[39m# then it's still \"all ASCII\" even though the domain name has\u001b[39;00m\n\u001b[0;32m    971\u001b[0m \u001b[39m# codepoints > 127.\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\name.py:925\u001b[0m, in \u001b[0;36mfrom_unicode\u001b[1;34m(text, origin, idna_codec)\u001b[0m\n\u001b[0;32m    924\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(label) \u001b[39m>\u001b[39m \u001b[39m0\u001b[39m:\n\u001b[1;32m--> 925\u001b[0m     labels\u001b[39m.\u001b[39mappend(idna_codec\u001b[39m.\u001b[39;49mencode(label))\n\u001b[0;32m    926\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\name.py:213\u001b[0m, in \u001b[0;36mIDNA2003Codec.encode\u001b[1;34m(self, label)\u001b[0m\n\u001b[0;32m    212\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mUnicodeError\u001b[39;00m:\n\u001b[1;32m--> 213\u001b[0m     \u001b[39mraise\u001b[39;00m LabelTooLong\n",
      "\u001b[1;31mLabelTooLong\u001b[0m: A DNS label is > 63 octets long.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mConfigurationError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[2], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mpymongo\u001b[39;00m \u001b[39mimport\u001b[39;00m MongoClient\n\u001b[1;32m----> 2\u001b[0m myclient \u001b[39m=\u001b[39m pymongo\u001b[39m.\u001b[39;49mMongoClient(\u001b[39m\"\u001b[39;49m\u001b[39mmongodb+srv://mnazrth:jX7krpmbpVa3I3GN@cluster0.gwghjgh.mongodb.net/?retryWrites=true&w=majority\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n\u001b[0;32m      3\u001b[0m \u001b[39m# myclient = pymongo.MongoClient(\"mongodb://3.112.247.246:27017/\")\u001b[39;00m\n\u001b[0;32m      4\u001b[0m \n\u001b[0;32m      5\u001b[0m \n\u001b[0;32m      6\u001b[0m \u001b[39m# myclient = MongoClient('localhost', 27017)\u001b[39;00m\n\u001b[0;32m      7\u001b[0m \n\u001b[0;32m      8\u001b[0m \u001b[39m# myclient = pymongo.MongoClient(\"mongodb+srv://mnazrth:jX7krpmbpVa3I3GN@cluster0.gwghjgh.mongodb.net/?retryWrites=true&w=majority\")\u001b[39;00m\n\u001b[0;32m      9\u001b[0m mydb \u001b[39m=\u001b[39m myclient[\u001b[39m\"\u001b[39m\u001b[39mtest\u001b[39m\u001b[39m\"\u001b[39m]\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\mongo_client.py:736\u001b[0m, in \u001b[0;36mMongoClient.__init__\u001b[1;34m(self, host, port, document_class, tz_aware, connect, type_registry, **kwargs)\u001b[0m\n\u001b[0;32m    732\u001b[0m \u001b[39mif\u001b[39;00m timeout \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[0;32m    733\u001b[0m     timeout \u001b[39m=\u001b[39m common\u001b[39m.\u001b[39mvalidate_timeout_or_none_or_zero(\n\u001b[0;32m    734\u001b[0m         keyword_opts\u001b[39m.\u001b[39mcased_key(\u001b[39m\"\u001b[39m\u001b[39mconnecttimeoutms\u001b[39m\u001b[39m\"\u001b[39m), timeout\n\u001b[0;32m    735\u001b[0m     )\n\u001b[1;32m--> 736\u001b[0m res \u001b[39m=\u001b[39m uri_parser\u001b[39m.\u001b[39;49mparse_uri(\n\u001b[0;32m    737\u001b[0m     entity,\n\u001b[0;32m    738\u001b[0m     port,\n\u001b[0;32m    739\u001b[0m     validate\u001b[39m=\u001b[39;49m\u001b[39mTrue\u001b[39;49;00m,\n\u001b[0;32m    740\u001b[0m     warn\u001b[39m=\u001b[39;49m\u001b[39mTrue\u001b[39;49;00m,\n\u001b[0;32m    741\u001b[0m     normalize\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[0;32m    742\u001b[0m     connect_timeout\u001b[39m=\u001b[39;49mtimeout,\n\u001b[0;32m    743\u001b[0m     srv_service_name\u001b[39m=\u001b[39;49msrv_service_name,\n\u001b[0;32m    744\u001b[0m     srv_max_hosts\u001b[39m=\u001b[39;49msrv_max_hosts,\n\u001b[0;32m    745\u001b[0m )\n\u001b[0;32m    746\u001b[0m seeds\u001b[39m.\u001b[39mupdate(res[\u001b[39m\"\u001b[39m\u001b[39mnodelist\u001b[39m\u001b[39m\"\u001b[39m])\n\u001b[0;32m    747\u001b[0m username \u001b[39m=\u001b[39m res[\u001b[39m\"\u001b[39m\u001b[39musername\u001b[39m\u001b[39m\"\u001b[39m] \u001b[39mor\u001b[39;00m username\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\uri_parser.py:542\u001b[0m, in \u001b[0;36mparse_uri\u001b[1;34m(uri, default_port, validate, warn, normalize, connect_timeout, srv_service_name, srv_max_hosts)\u001b[0m\n\u001b[0;32m    540\u001b[0m connect_timeout \u001b[39m=\u001b[39m connect_timeout \u001b[39mor\u001b[39;00m options\u001b[39m.\u001b[39mget(\u001b[39m\"\u001b[39m\u001b[39mconnectTimeoutMS\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m    541\u001b[0m dns_resolver \u001b[39m=\u001b[39m _SrvResolver(fqdn, connect_timeout, srv_service_name, srv_max_hosts)\n\u001b[1;32m--> 542\u001b[0m nodes \u001b[39m=\u001b[39m dns_resolver\u001b[39m.\u001b[39;49mget_hosts()\n\u001b[0;32m    543\u001b[0m dns_options \u001b[39m=\u001b[39m dns_resolver\u001b[39m.\u001b[39mget_options()\n\u001b[0;32m    544\u001b[0m \u001b[39mif\u001b[39;00m dns_options:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:121\u001b[0m, in \u001b[0;36m_SrvResolver.get_hosts\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    120\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mget_hosts\u001b[39m(\u001b[39mself\u001b[39m):\n\u001b[1;32m--> 121\u001b[0m     _, nodes \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_get_srv_response_and_hosts(\u001b[39mTrue\u001b[39;49;00m)\n\u001b[0;32m    122\u001b[0m     \u001b[39mreturn\u001b[39;00m nodes\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:101\u001b[0m, in \u001b[0;36m_SrvResolver._get_srv_response_and_hosts\u001b[1;34m(self, encapsulate_errors)\u001b[0m\n\u001b[0;32m    100\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_get_srv_response_and_hosts\u001b[39m(\u001b[39mself\u001b[39m, encapsulate_errors):\n\u001b[1;32m--> 101\u001b[0m     results \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_resolve_uri(encapsulate_errors)\n\u001b[0;32m    103\u001b[0m     \u001b[39m# Construct address tuples\u001b[39;00m\n\u001b[0;32m    104\u001b[0m     nodes \u001b[39m=\u001b[39m [\n\u001b[0;32m    105\u001b[0m         (maybe_decode(res\u001b[39m.\u001b[39mtarget\u001b[39m.\u001b[39mto_text(omit_final_dot\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m)), res\u001b[39m.\u001b[39mport) \u001b[39mfor\u001b[39;00m res \u001b[39min\u001b[39;00m results\n\u001b[0;32m    106\u001b[0m     ]\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:97\u001b[0m, in \u001b[0;36m_SrvResolver._resolve_uri\u001b[1;34m(self, encapsulate_errors)\u001b[0m\n\u001b[0;32m     95\u001b[0m         \u001b[39mraise\u001b[39;00m\n\u001b[0;32m     96\u001b[0m     \u001b[39m# Else, raise all errors as ConfigurationError.\u001b[39;00m\n\u001b[1;32m---> 97\u001b[0m     \u001b[39mraise\u001b[39;00m ConfigurationError(\u001b[39mstr\u001b[39m(exc))\n\u001b[0;32m     98\u001b[0m \u001b[39mreturn\u001b[39;00m results\n",
      "\u001b[1;31mConfigurationError\u001b[0m: A DNS label is > 63 octets long."
     ]
    }
   ],
   "source": [
    "from pymongo import MongoClient\n",
    "myclient = pymongo.MongoClient(\"mongodb+srv://mnazrth:jX7krpmbpVa3I3GN@cluster0.gwghjgh.mongodb.net/?retryWrites=true&w=majority\")\n",
    "\n",
    "mydb = myclient[\"test\"]\n",
    "mycol = mydb[\"users\"]\n",
    "\n",
    "# Check the connection status\n",
    "if myclient.is_mongos:\n",
    "    print(\"Connected to MongoDB as a mongos router.\")\n",
    "elif myclient.is_primary:\n",
    "    print(\"Connected to MongoDB as a primary node.\")\n",
    "elif myclient.is_secondary:\n",
    "    print(\"Connected to MongoDB as a secondary node.\")\n",
    "else:\n",
    "    print(\"Connection status: Unknown.\")\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "reservation = {  \"license_plate\":\"75062834\" }\n",
    "parking_spot={'status':'true'}\n",
    "x = mycol.insert_one(reservation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_id': ObjectId('64b56ef6f64ee9fcb7aec32d'), 'license_plate': '75062834'}\n"
     ]
    }
   ],
   "source": [
    "for x in mycol.find():\n",
    "      print(x)\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "ConfigurationError",
     "evalue": "A DNS label is > 63 octets long.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mUnicodeError\u001b[0m                              Traceback (most recent call last)",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\name.py:211\u001b[0m, in \u001b[0;36mIDNA2003Codec.encode\u001b[1;34m(self, label)\u001b[0m\n\u001b[0;32m    210\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m--> 211\u001b[0m     \u001b[39mreturn\u001b[39;00m encodings\u001b[39m.\u001b[39;49midna\u001b[39m.\u001b[39;49mToASCII(label)\n\u001b[0;32m    212\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mUnicodeError\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\encodings\\idna.py:74\u001b[0m, in \u001b[0;36mToASCII\u001b[1;34m(label)\u001b[0m\n\u001b[0;32m     73\u001b[0m \u001b[39m# Step 2: nameprep\u001b[39;00m\n\u001b[1;32m---> 74\u001b[0m label \u001b[39m=\u001b[39m nameprep(label)\n\u001b[0;32m     76\u001b[0m \u001b[39m# Step 3: UseSTD3ASCIIRules is false\u001b[39;00m\n\u001b[0;32m     77\u001b[0m \u001b[39m# Step 4: try ASCII\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\encodings\\idna.py:50\u001b[0m, in \u001b[0;36mnameprep\u001b[1;34m(label)\u001b[0m\n\u001b[0;32m     49\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39many\u001b[39m(stringprep\u001b[39m.\u001b[39min_table_d2(x) \u001b[39mfor\u001b[39;00m x \u001b[39min\u001b[39;00m label):\n\u001b[1;32m---> 50\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mUnicodeError\u001b[39;00m(\u001b[39m\"\u001b[39m\u001b[39mViolation of BIDI requirement 2\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m     51\u001b[0m \u001b[39m# 3) If a string contains any RandALCat character, a\u001b[39;00m\n\u001b[0;32m     52\u001b[0m \u001b[39m# RandALCat character MUST be the first character of the\u001b[39;00m\n\u001b[0;32m     53\u001b[0m \u001b[39m# string, and a RandALCat character MUST be the last\u001b[39;00m\n\u001b[0;32m     54\u001b[0m \u001b[39m# character of the string.\u001b[39;00m\n",
      "\u001b[1;31mUnicodeError\u001b[0m: Violation of BIDI requirement 2",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mLabelTooLong\u001b[0m                              Traceback (most recent call last)",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:89\u001b[0m, in \u001b[0;36m_SrvResolver._resolve_uri\u001b[1;34m(self, encapsulate_errors)\u001b[0m\n\u001b[0;32m     88\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m---> 89\u001b[0m     results \u001b[39m=\u001b[39m _resolve(\n\u001b[0;32m     90\u001b[0m         \u001b[39m\"\u001b[39;49m\u001b[39m_\u001b[39;49m\u001b[39m\"\u001b[39;49m \u001b[39m+\u001b[39;49m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__srv \u001b[39m+\u001b[39;49m \u001b[39m\"\u001b[39;49m\u001b[39m._tcp.\u001b[39;49m\u001b[39m\"\u001b[39;49m \u001b[39m+\u001b[39;49m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__fqdn, \u001b[39m\"\u001b[39;49m\u001b[39mSRV\u001b[39;49m\u001b[39m\"\u001b[39;49m, lifetime\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__connect_timeout\n\u001b[0;32m     91\u001b[0m     )\n\u001b[0;32m     92\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mException\u001b[39;00m \u001b[39mas\u001b[39;00m exc:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:43\u001b[0m, in \u001b[0;36m_resolve\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m     41\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mhasattr\u001b[39m(resolver, \u001b[39m\"\u001b[39m\u001b[39mresolve\u001b[39m\u001b[39m\"\u001b[39m):\n\u001b[0;32m     42\u001b[0m     \u001b[39m# dnspython >= 2\u001b[39;00m\n\u001b[1;32m---> 43\u001b[0m     \u001b[39mreturn\u001b[39;00m resolver\u001b[39m.\u001b[39;49mresolve(\u001b[39m*\u001b[39;49margs, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n\u001b[0;32m     44\u001b[0m \u001b[39m# dnspython 1.X\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:1368\u001b[0m, in \u001b[0;36mresolve\u001b[1;34m(qname, rdtype, rdclass, tcp, source, raise_on_no_answer, source_port, lifetime, search)\u001b[0m\n\u001b[0;32m   1359\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"Query nameservers to find the answer to the question.\u001b[39;00m\n\u001b[0;32m   1360\u001b[0m \n\u001b[0;32m   1361\u001b[0m \u001b[39mThis is a convenience function that uses the default resolver\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1365\u001b[0m \u001b[39mparameters.\u001b[39;00m\n\u001b[0;32m   1366\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m-> 1368\u001b[0m \u001b[39mreturn\u001b[39;00m get_default_resolver()\u001b[39m.\u001b[39mresolve(\n\u001b[0;32m   1369\u001b[0m     qname,\n\u001b[0;32m   1370\u001b[0m     rdtype,\n\u001b[0;32m   1371\u001b[0m     rdclass,\n\u001b[0;32m   1372\u001b[0m     tcp,\n\u001b[0;32m   1373\u001b[0m     source,\n\u001b[0;32m   1374\u001b[0m     raise_on_no_answer,\n\u001b[0;32m   1375\u001b[0m     source_port,\n\u001b[0;32m   1376\u001b[0m     lifetime,\n\u001b[0;32m   1377\u001b[0m     search,\n\u001b[0;32m   1378\u001b[0m )\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:1331\u001b[0m, in \u001b[0;36mget_default_resolver\u001b[1;34m()\u001b[0m\n\u001b[0;32m   1330\u001b[0m \u001b[39mif\u001b[39;00m default_resolver \u001b[39mis\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m-> 1331\u001b[0m     reset_default_resolver()\n\u001b[0;32m   1332\u001b[0m \u001b[39massert\u001b[39;00m default_resolver \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:1344\u001b[0m, in \u001b[0;36mreset_default_resolver\u001b[1;34m()\u001b[0m\n\u001b[0;32m   1343\u001b[0m \u001b[39mglobal\u001b[39;00m default_resolver\n\u001b[1;32m-> 1344\u001b[0m default_resolver \u001b[39m=\u001b[39m Resolver()\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:856\u001b[0m, in \u001b[0;36mBaseResolver.__init__\u001b[1;34m(self, filename, configure)\u001b[0m\n\u001b[0;32m    845\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"*filename*, a ``str`` or file object, specifying a file\u001b[39;00m\n\u001b[0;32m    846\u001b[0m \u001b[39min standard /etc/resolv.conf format.  This parameter is meaningful\u001b[39;00m\n\u001b[0;32m    847\u001b[0m \u001b[39monly when *configure* is true and the platform is POSIX.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    853\u001b[0m \u001b[39mon Windows systems.)\u001b[39;00m\n\u001b[0;32m    854\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m--> 856\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mreset()\n\u001b[0;32m    857\u001b[0m \u001b[39mif\u001b[39;00m configure:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\resolver.py:866\u001b[0m, in \u001b[0;36mBaseResolver.reset\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    864\u001b[0m \u001b[39m\u001b[39m\u001b[39m\"\"\"Reset all resolver configuration to the defaults.\"\"\"\u001b[39;00m\n\u001b[1;32m--> 866\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mdomain \u001b[39m=\u001b[39m dns\u001b[39m.\u001b[39mname\u001b[39m.\u001b[39mName(dns\u001b[39m.\u001b[39;49mname\u001b[39m.\u001b[39;49mfrom_text(socket\u001b[39m.\u001b[39;49mgethostname())[\u001b[39m1\u001b[39m:])\n\u001b[0;32m    867\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mdomain) \u001b[39m==\u001b[39m \u001b[39m0\u001b[39m:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\name.py:963\u001b[0m, in \u001b[0;36mfrom_text\u001b[1;34m(text, origin, idna_codec)\u001b[0m\n\u001b[0;32m    961\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m is_all_ascii(text):\n\u001b[0;32m    962\u001b[0m     \u001b[39m# Some codepoint in the input text is > 127, so IDNA applies.\u001b[39;00m\n\u001b[1;32m--> 963\u001b[0m     \u001b[39mreturn\u001b[39;00m from_unicode(text, origin, idna_codec)\n\u001b[0;32m    964\u001b[0m \u001b[39m# The input is all ASCII, so treat this like an ordinary non-IDNA\u001b[39;00m\n\u001b[0;32m    965\u001b[0m \u001b[39m# domain name.  Note that \"all ASCII\" is about the input text,\u001b[39;00m\n\u001b[0;32m    966\u001b[0m \u001b[39m# not the codepoints in the domain name.  E.g. if text has value\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    970\u001b[0m \u001b[39m# then it's still \"all ASCII\" even though the domain name has\u001b[39;00m\n\u001b[0;32m    971\u001b[0m \u001b[39m# codepoints > 127.\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\name.py:925\u001b[0m, in \u001b[0;36mfrom_unicode\u001b[1;34m(text, origin, idna_codec)\u001b[0m\n\u001b[0;32m    924\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(label) \u001b[39m>\u001b[39m \u001b[39m0\u001b[39m:\n\u001b[1;32m--> 925\u001b[0m     labels\u001b[39m.\u001b[39mappend(idna_codec\u001b[39m.\u001b[39;49mencode(label))\n\u001b[0;32m    926\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\dns\\name.py:213\u001b[0m, in \u001b[0;36mIDNA2003Codec.encode\u001b[1;34m(self, label)\u001b[0m\n\u001b[0;32m    212\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mUnicodeError\u001b[39;00m:\n\u001b[1;32m--> 213\u001b[0m     \u001b[39mraise\u001b[39;00m LabelTooLong\n",
      "\u001b[1;31mLabelTooLong\u001b[0m: A DNS label is > 63 octets long.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mConfigurationError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[10], line 7\u001b[0m\n\u001b[0;32m      4\u001b[0m connection_string \u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mmongodb+srv://mnazrth:jX7krpmbpVa3I3GN@cluster0.gwghjgh.mongodb.net/?retryWrites=true&w=majority\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m      6\u001b[0m \u001b[39m# Create a MongoDB client\u001b[39;00m\n\u001b[1;32m----> 7\u001b[0m myclient \u001b[39m=\u001b[39m MongoClient(connection_string)\n\u001b[0;32m      9\u001b[0m \u001b[39m# Access the database\u001b[39;00m\n\u001b[0;32m     10\u001b[0m \u001b[39m# mydb = myclient[\"mydatabase\"]\u001b[39;00m\n\u001b[0;32m     11\u001b[0m \u001b[39m# mycol = mydb[\"users\"]\u001b[39;00m\n\u001b[0;32m     12\u001b[0m mydb \u001b[39m=\u001b[39m myclient[\u001b[39m'\u001b[39m\u001b[39mmydatabase\u001b[39m\u001b[39m'\u001b[39m]  \u001b[39m# Replace 'your_database_name' with the actual name of your database\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\mongo_client.py:736\u001b[0m, in \u001b[0;36mMongoClient.__init__\u001b[1;34m(self, host, port, document_class, tz_aware, connect, type_registry, **kwargs)\u001b[0m\n\u001b[0;32m    732\u001b[0m \u001b[39mif\u001b[39;00m timeout \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[0;32m    733\u001b[0m     timeout \u001b[39m=\u001b[39m common\u001b[39m.\u001b[39mvalidate_timeout_or_none_or_zero(\n\u001b[0;32m    734\u001b[0m         keyword_opts\u001b[39m.\u001b[39mcased_key(\u001b[39m\"\u001b[39m\u001b[39mconnecttimeoutms\u001b[39m\u001b[39m\"\u001b[39m), timeout\n\u001b[0;32m    735\u001b[0m     )\n\u001b[1;32m--> 736\u001b[0m res \u001b[39m=\u001b[39m uri_parser\u001b[39m.\u001b[39;49mparse_uri(\n\u001b[0;32m    737\u001b[0m     entity,\n\u001b[0;32m    738\u001b[0m     port,\n\u001b[0;32m    739\u001b[0m     validate\u001b[39m=\u001b[39;49m\u001b[39mTrue\u001b[39;49;00m,\n\u001b[0;32m    740\u001b[0m     warn\u001b[39m=\u001b[39;49m\u001b[39mTrue\u001b[39;49;00m,\n\u001b[0;32m    741\u001b[0m     normalize\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[0;32m    742\u001b[0m     connect_timeout\u001b[39m=\u001b[39;49mtimeout,\n\u001b[0;32m    743\u001b[0m     srv_service_name\u001b[39m=\u001b[39;49msrv_service_name,\n\u001b[0;32m    744\u001b[0m     srv_max_hosts\u001b[39m=\u001b[39;49msrv_max_hosts,\n\u001b[0;32m    745\u001b[0m )\n\u001b[0;32m    746\u001b[0m seeds\u001b[39m.\u001b[39mupdate(res[\u001b[39m\"\u001b[39m\u001b[39mnodelist\u001b[39m\u001b[39m\"\u001b[39m])\n\u001b[0;32m    747\u001b[0m username \u001b[39m=\u001b[39m res[\u001b[39m\"\u001b[39m\u001b[39musername\u001b[39m\u001b[39m\"\u001b[39m] \u001b[39mor\u001b[39;00m username\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\uri_parser.py:542\u001b[0m, in \u001b[0;36mparse_uri\u001b[1;34m(uri, default_port, validate, warn, normalize, connect_timeout, srv_service_name, srv_max_hosts)\u001b[0m\n\u001b[0;32m    540\u001b[0m connect_timeout \u001b[39m=\u001b[39m connect_timeout \u001b[39mor\u001b[39;00m options\u001b[39m.\u001b[39mget(\u001b[39m\"\u001b[39m\u001b[39mconnectTimeoutMS\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m    541\u001b[0m dns_resolver \u001b[39m=\u001b[39m _SrvResolver(fqdn, connect_timeout, srv_service_name, srv_max_hosts)\n\u001b[1;32m--> 542\u001b[0m nodes \u001b[39m=\u001b[39m dns_resolver\u001b[39m.\u001b[39;49mget_hosts()\n\u001b[0;32m    543\u001b[0m dns_options \u001b[39m=\u001b[39m dns_resolver\u001b[39m.\u001b[39mget_options()\n\u001b[0;32m    544\u001b[0m \u001b[39mif\u001b[39;00m dns_options:\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:121\u001b[0m, in \u001b[0;36m_SrvResolver.get_hosts\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    120\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mget_hosts\u001b[39m(\u001b[39mself\u001b[39m):\n\u001b[1;32m--> 121\u001b[0m     _, nodes \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_get_srv_response_and_hosts(\u001b[39mTrue\u001b[39;49;00m)\n\u001b[0;32m    122\u001b[0m     \u001b[39mreturn\u001b[39;00m nodes\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:101\u001b[0m, in \u001b[0;36m_SrvResolver._get_srv_response_and_hosts\u001b[1;34m(self, encapsulate_errors)\u001b[0m\n\u001b[0;32m    100\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_get_srv_response_and_hosts\u001b[39m(\u001b[39mself\u001b[39m, encapsulate_errors):\n\u001b[1;32m--> 101\u001b[0m     results \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_resolve_uri(encapsulate_errors)\n\u001b[0;32m    103\u001b[0m     \u001b[39m# Construct address tuples\u001b[39;00m\n\u001b[0;32m    104\u001b[0m     nodes \u001b[39m=\u001b[39m [\n\u001b[0;32m    105\u001b[0m         (maybe_decode(res\u001b[39m.\u001b[39mtarget\u001b[39m.\u001b[39mto_text(omit_final_dot\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m)), res\u001b[39m.\u001b[39mport) \u001b[39mfor\u001b[39;00m res \u001b[39min\u001b[39;00m results\n\u001b[0;32m    106\u001b[0m     ]\n",
      "File \u001b[1;32mc:\\Users\\Asus\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pymongo\\srv_resolver.py:97\u001b[0m, in \u001b[0;36m_SrvResolver._resolve_uri\u001b[1;34m(self, encapsulate_errors)\u001b[0m\n\u001b[0;32m     95\u001b[0m         \u001b[39mraise\u001b[39;00m\n\u001b[0;32m     96\u001b[0m     \u001b[39m# Else, raise all errors as ConfigurationError.\u001b[39;00m\n\u001b[1;32m---> 97\u001b[0m     \u001b[39mraise\u001b[39;00m ConfigurationError(\u001b[39mstr\u001b[39m(exc))\n\u001b[0;32m     98\u001b[0m \u001b[39mreturn\u001b[39;00m results\n",
      "\u001b[1;31mConfigurationError\u001b[0m: A DNS label is > 63 octets long."
     ]
    }
   ],
   "source": [
    "from pymongo import MongoClient\n",
    "\n",
    "# Set up the MongoDB connection string\n",
    "connection_string = \"mongodb+srv://mnazrth:jX7krpmbpVa3I3GN@cluster0.gwghjgh.mongodb.net/?retryWrites=true&w=majority\"\n",
    "\n",
    "# Create a MongoDB client\n",
    "myclient = MongoClient(connection_string)\n",
    "\n",
    "# Access the database\n",
    "# mydb = myclient[\"mydatabase\"]\n",
    "# mycol = mydb[\"users\"]\n",
    "mydb = myclient['test']  # Replace 'your_database_name' with the actual name of your database\n",
    "\n",
    "# Access a collection within the database\n",
    "mycol = mydb['users']  # Replace 'your_collection_name' with the actual name of your collection\n",
    "\n",
    "# Perform database operations\n",
    "# For example, you can query documents from the collection\n",
    "documents = collection.find()\n",
    "\n",
    "for document in documents:\n",
    "    print(document)\n",
    "\n",
    "# Close the connection\n",
    "myclient.close()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "369f2c481f4da34e4445cda3fffd2e751bd1c4d706f27375911949ba6bb62e1c"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
